packages/core/src/modules/connections/handlers/ConnectionRequestHandler.ts

```diff
@@ -32,16 +34,23 @@ export class ConnectionRequestHandler implements MessageHandler {
   }
 
   public async handle(messageContext: MessageHandlerInboundMessage<ConnectionRequestHandler>) {
-    const { connection, recipientKey, senderKey } = messageContext
+    const { agentContext, connection, recipientKey, senderKey, message } = messageContext
 
     if (!recipientKey || !senderKey) {
       throw new AriesFrameworkError('Unable to process connection request without senderVerkey or recipientKey')
     }
 
-    const outOfBandRecord = await this.outOfBandService.findCreatedByRecipientKey(
-      messageContext.agentContext,
-      recipientKey
-    )
+    const parentThreadId = message.thread?.parentThreadId
+
+    const outOfBandRecord =
+      parentThreadId && tryParseDid(parentThreadId)
+        ? await this.outOfBandService.createImplicitInvitation(agentContext, {
```

Comment:
I'm not 100% on the naming of this method. We've just received a request based on an implicit invitation (the service in the did document is the implicit invitation). It feels like `createFromImplicitInvitation` would be more suitable? Also not 100% happy with that, but it should denote that we're creating an out of band record based on a request that was sent based on an implicit invitation.

Reply:
Well, at least I removed that `isImplicitInvitation` that didn't convince you neither 😆. The naming was somehow borrowed from the OutOfBandApi method called `createInvitation` that returned an `OutOfBandRecord`. But you are completely right: `createFromImplicitInvitation` is a bit more accurate. Could be also `createFromConnectionRequest` or `createFromImplicitInvitationResponse`? I still prefer your suggestion.